hiyapyco/__init__.py

```diff
@@ -119,6 +124,19 @@ def __init__(self, *args, **kwargs):
             self.mergelists = kwargs['mergelists']
             del kwargs['mergelists']
 
+        self.none_behavior = None
+        if "none_behavior" in kwargs:
+            logger.debug("parse kwarg method: %s ..." % kwargs["none_behavior"])
+            if kwargs["none_behavior"] not in NONE_BEHAVIORS.values():
+                raise HiYaPyCoInvocationException(
+                    "undefined method used, must be one of: %s"
+                    % " ".join(NONE_BEHAVIORS.keys())
+                )
+            self.none_behavior = kwargs["none_behavior"]
+            del kwargs["none_behavior"]
+        if self.none_behavior is None:
+            self.none_behavior = NONE_BEHAVIOR_DEFAULT
```

Comment:
Is there a reason why arguments are evaulated in such a (IMO) cumbersome way? This could be just a keyword argument to the function instead of manually fiddling with `kwargs`... I would probably have implemented it like that when starting from scratch:

```python
NoneBehavior = Enum("NoneBehavior", ["DEFAULT", "OVERRIDE"])

def __init__(self, *args, none_behavior: NoneBehavior = None, **kwargs):
    if none_behavior not in NoneBehavior:
        raise HiYaPyCoInvocationException(f"Undefined non_behavior, must be one of {', '.join(NoneBehavior._member_names_}")
    self.none_behavior = none_behavior
```

Reply:
@grst ... I see what you mean, but to be honest I can not remember what idea or cerebral nod motivated me to implement it in such a detailed and expansive notation approx 10 years ago ... to be honest, I had no idea about the widespread use and longevity of the project during the first implementation that was more intended for my private use
Anyhow, just to not let the topic drown, I opened #81, thank you for your hint.

Reply:
10 years is also a long time ago and many of the newer language features like type hints or enums didn't exist. But nowadays Python 3.9 is the [oldest supported version](https://devguide.python.org/versions/) :)